### PR TITLE
Move gamelift client usage into each activity scope

### PR DESCRIPTION
**Pull request by onecent1101**
## Details
Previously we move gamelift client ownership to the system component, so we don't need to pass it to each activity. Instead we can get it in activity scope directly.

## Testing
Pass unit test

Signed-off-by: onecent1101 <liug@amazon.com>

### CODE REVIEW

File: Gems/AWSGameLift/Code/AWSGameLiftClient/Source/Activity/AWSGameLiftCreateSessionActivity.cpp [hunk 1]
```diff
@@ -63,15 +70,21 @@ namespace AWSGameLift
...
         }
 
-        AZStd::string CreateSession(
-            const Aws::GameLift::GameLiftClient& gameliftClient,
-            const AWSGameLiftCreateSessionRequest& createSessionRequest)
+        AZStd::string CreateSession(const AWSGameLiftCreateSessionRequest& createSessionRequest)
         {
+            AZStd::string result = "";
+
+            auto gameliftClient = AZ::Interface<IAWSGameLiftInternalRequests>::Get()->GetGameLiftClient();
```
Comment (lmbr-pip) [inline Gems/AWSGameLift/Code/AWSGameLiftClient/Source/Activity/AWSGameLiftCreateSessionActivity.cpp:77]: DRY: For another PR, this same code is in a lot of places (same code block to get and validate client). IMHO we should reduce bloat like this.

File: Gems/AWSGameLift/Code/AWSGameLiftClient/Source/Activity/AWSGameLiftCreateSessionOnQueueActivity.h [hunk 2]
```diff
@@ -25,9 +23,7 @@ namespace AWSGameLift
             const AWSGameLiftCreateSessionOnQueueRequest& createSessionOnQueueRequest);
 
         // Create StartGameSessionPlacementRequest and make a CreateGameSession call through GameLift client
-        AZStd::string CreateSessionOnQueue(
-            const Aws::GameLift::GameLiftClient& gameliftClient,
-            const AWSGameLiftCreateSessionOnQueueRequest& createSessionOnQueueRequest);
+        AZStd::string CreateSessionOnQueue(const AWSGameLiftCreateSessionOnQueueRequest& createSessionOnQueueRequest);
```
Comment (lmbr-pip) [inline Gems/AWSGameLift/Code/AWSGameLiftClient/Source/Activity/AWSGameLiftCreateSessionOnQueueActivity.h:26]: minor: Maybe order is Build, Validate then Create ?

File: Gems/AWSGameLift/Code/AWSGameLiftClient/Source/Activity/AWSGameLiftCreateSessionActivity.cpp (hunk 1 above)
Comment (onecent1101) [reply Gems/AWSGameLift/Code/AWSGameLiftClient/Source/Activity/AWSGameLiftCreateSessionActivity.cpp:77]: I am not sure how to dedupe further here, and what is the value of doing it.
```
AZStd::shared_ptr<GameLiftClient> GetGameLiftClient(const AZStd::string& logTag)
{
    if (m_gameliftClient) {
        return m_gameliftClient;
    } else {
        AZ_Error(logTag, false, AWSGameLiftClientMissingErrorMessage);
        return nullptr;
    }
}

// caller
...
AZStd::string result = "";
auto gameliftClient = AZ::Interface<IAWSGameLiftInternalRequests>::Get()->GetGameLiftClient();
if (!gameliftClient)
{
    return result;
}
...
```
With above thing, you just dedupe the `AZ_Error` here. And most importantly, you make assumption here that client getter function should always log error when it is nullptr. 
You can optimize more, like pass a lambda function as parameter to this getter function so caller has power to control the log, but the weirdest thing is why do we even need a logging lambda function as client getter function parameter.

**Review by junbo75 — APPROVED** [review]
LG. Assume that we will have another round of refactor tracked by a separate PR.

**Comment by onecent1101** [comment]
> LG. Assume that we will have another round of refactor tracked by a separate PR.

This PR is not for refactor, just a cleanup leftover from gamelift flexmatch. Refactor is tracked by a separate task.

**Review by lmbr-pip — APPROVED** [review]

**Comment by rainbj** [comment]
Hi @allisaurus and @junbo75 does this activity have any UX related changes?

**Comment by onecent1101** [comment]
> Hi @allisaurus and @junbo75 does this activity have any UX related changes?

Didn't realize there is a question, thanks for @allisaurus notification. 
There is no UX change.

**Pull request merged**